management/src/main/java/io/micronaut/management/endpoint/management/ManagementLinkCollector.java

```diff
@@ -0,0 +1,17 @@
+package io.micronaut.management.endpoint.management;
+
+import io.micronaut.http.hateoas.Resource;
+import io.micronaut.management.endpoint.management.impl.ManagementResponse;
+import io.micronaut.web.router.UriRoute;
+
+import java.util.stream.Stream;
+
+/**
+ * <p>Collect routes in a {@link Resource} to respond in {@link ManagementEndpoint}.</p>
+ *
+ * @author Hernán Cervera
+ * @since 2.5
+ * */
+public interface ManagementLinkCollector {
+    Resource collectLinks(Stream<UriRoute> routes, String selfEndpointId);
```

Comment:
Similar to above, the interface shouldn't assume `Resource` or `links`. This should also return a `Publisher`

Reply:
Now it returns a `Publisher`